lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -39,26 +39,93 @@ def object
               object
             end
           end
+
+          ResourceIdentifier = Struct.new(:id, :type) do
```

Comment:
cool.

so what is a `ResourceIdentifier` what does it represent?
(I think your answer to this should be coupled with inline comments as class header comments)